src/lib/transport.js

```diff
@@ -0,0 +1,223 @@
+'use strict'
+
+const Packet = require('../utils/packet')
+const moment = require('moment')
+const cryptoHelper = require('../utils/crypto')
+const cc = require('../utils/condition')
+const co = require('co')
+const debug = require('debug')('ilp:transport')
+const assert = require('assert')
+const base64url = require('../utils/base64url')
+const BigNumber = require('bignumber.js')
+const { safeConnect } = require('../utils')
+
+function _safeDecrypt (data, secret) {
+  if (!data) return {}
+  try {
+    return cryptoHelper.aesDecryptObject(data, secret)
+  } catch (err) {
+    debug('decryption error:', err.message)
+    return undefined
+  }
+}
+
+function createPacketAndCondition ({
+  id,
+  destinationAmount,
+  destinationAccount,
+  secret,
+  data,
+  expiresAt
+}, protocol) {
+  assert(typeof destinationAmount === 'string', 'destinationAmount must be a string')
+  assert(typeof destinationAccount === 'string', 'destinationAccount must be a string')
+  assert(Buffer.isBuffer(secret), 'secret must be a buffer')
+
+  const receiverId = base64url(cryptoHelper.getReceiverId(secret))
+  const address = destinationAccount + '.~' + protocol + '.' + receiverId +
+    (id ? ('.' + id) : '')
+
+  const blobData = { expiresAt, data }
+  const blob = data &&
```

Comment:
this is a bug; if data is undefined, that doesn't mean blob has to be. at least that's what you just said yourself ;)